### DISCUSSION TITLE
How to load a model dynamic library?

### DISCUSSION THREAD

**Discussion by Zachary9079** [SanderMertens/flecs#486] [Q&A]
Use example_ cpp_ 04_ simple_ Module generates a dynamic library, and then uses the "ecs_import_from_library" interface to import the model dynamic library. The error "library loading not supported, set module_to_dl, dlopen, dlclose and dlproc OS API calls first" will be reported

**Comment by codylico** [comment]
_(warning: untested code follows)_
It sounds like the `flecs` library wants you to configure the operating system interface (i.e. the "OS API"), which is like a big virtual table of functions that do platform dependent stuff. Such configuration may look like the following:

```c++
ecs_os_dl_t my_api_dlopen(const char *libname) {
#ifdef _WIN32
  return LoadLibraryA(libname);
#elif (defined __linux__)
  return dlopen(libname, RTLD_NOW|RTLD_LOCAL);
#else
  return NULL;
#endif
}

int main() {
  ecs_os_api_t my_api;
  /* ... */
  my_api.dlopen_ = &my_api_dlopen;
  /* ... */
  ecs_os_set_api(&my_api);

  /* then start using the ECS, for example: */
  ecs_world_t *world = ecs_init();
  /* ... */
```

However, my code above would override the whole OS API, and I seem to remember that the `flecs` C++ interface comes with an OS API. @SanderMertens, is that true, and if so, is it/should it be missing the dlopen/etc. functions?
